### PR TITLE
Add support for Hal style collection with Pagerfanta

### PR DESCRIPTION
**Pull request by baldurrensch**
Needs to be cleaned up, unit tests added.

### CODE REVIEW

File: Serializer/Handler/HalPagerfantaHandler.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            'type' => 'FSC\HateoasBundle\Model\HalPagerfanta',
+            'method' => 'serializeToArray',
+        )
+        );
+    }
+
+    protected $serializerMetadataFactory;
+    protected $embedderEventSubscriber;
+    protected $linkEventSubscriber;
+    protected $xmlElementsNamesUseSerializerMetadata;
```
Comment (adrienbrault) [inline Serializer/Handler/HalPagerfantaHandler.php:34]: Can be removed

File: Serializer/Handler/HalPagerfantaHandler.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+use JMS\Serializer\XmlSerializationVisitor;
+use JMS\Serializer\GenericSerializationVisitor;
+use Metadata\MetadataFactoryInterface;
+
+use FSC\HateoasBundle\Model\HalPagerfanta;
+
+use FSC\HateoasBundle\Serializer\EventSubscriber\EmbedderEventSubscriber;
+use FSC\HateoasBundle\Serializer\EventSubscriber\LinkEventSubscriber;
+
+class HalPagerfantaHandler implements SubscribingHandlerInterface
```
Comment (adrienbrault) [inline Serializer/Handler/HalPagerfantaHandler.php:17]: Maybe it could also handle xml, and just serialize the pager.

**Comment by adrienbrault** [comment]
@baldurrensch See https://github.com/TheFootballSocialClub/FSCHateoasBundle/compare/master...hal_pager

**Comment by baldurrensch** [comment]
Nice. Looks good.

**Pull request closed without merging**